decidim-core/lib/decidim/icon_registry.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Decidim
+  class IconRegistry
+    def initialize
+      @icons = ActiveSupport::HashWithIndifferentAccess.new
+    end
+
+    # Registers a new icon.
+    #
+    # @param name [String] The name of the icon. It will be used to find the icon later.
+    # @param icon [String] The id of the icon. It will be used to load the icon from remixicon library.
```

Comment:
@andreslucena , I would not merge those. Having the separation, would allow integrator to actually replace any other icons as they please: 

in code we may have a call like: 
```
        Decidim.icons.register(name: "arrow-left-s-line", icon: "arrow-left-s-line", resource: "core", category: "system", description: "", engine: :core)
```

```
# within template
        <%= icon "arrow-left-s-line", class: "icon--small", role: "img", "aria-hidden": true %>
```

Then we would be able to perform this kind of override: 

```
        Decidim.icons.register(name: "arrow-left-s-line", icon: "corner-down-left-fill", resource: "core", category: "system", description: "", engine: :core)
#                                                                ^^^^^
```
Yes, i know that it may be counterintuitive, but this would allow us to avoid the template overrides.

Reply:
Also, what was previously defined as : 

```
DEFAULT_RESOURCE_TYPE_ICONS = {
  "Decidim::Proposals::CollaborativeDraft" => { icon: "draft-line", description: "Collaborative draft", category: "activity" },
  "Decidim::Proposals::Proposal" => { icon: "chat-new-line", description: "Proposal", category: "activity" },
  "Decidim::Amendment" => { icon: "git-branch-line", description: "Amendment", category: "activity" },
}
```

now is defined as : 

```
Decidim.icons.register(name: "Decidim::Proposals::CollaborativeDraft", icon: "draft-line", resource: "Decidim::Proposals::CollaborativeDraft", category: "activity", description: "Collaborative draft", engine: :proposals)
Decidim.icons.register(name: "Decidim::Proposals::Proposal", icon: "chat-new-line", resource: "Decidim::Proposals::Proposal", category: "activity", description: "Proposal", engine: :proposals)
Decidim.icons.register(name: "Decidim::Amendment", icon: "git-branch-line", resource: "Decidim::Amendment", category: "activity", description: "Amendment", engine: :core)
```

And that is being used inside of  to convert:
```
def resource_type_icon_key(resource_type)
  DEFAULT_RESOURCE_TYPE_ICONS.dig(resource_type.to_s, :icon) || DEFAULT_RESOURCE_TYPE_ICONS["other"][:icon]
end 
```
in 
```
def resource_type_icon_key(resource_type)
  Decidim.icons.find(resource_type.to_s).icon || Decidim.icons.find("other").icon
end
```

Reply:
Ok, with the IconHelper refactor it makes much more sense. 

Just to reiterate, but seeing this:

>         Decidim.icons.register(name: "arrow-left-s-line", icon: "arrow-left-s-line", resource: "core", category: "system", description: "", engine: :core)

It's too much verbosity and I actually thought that it was some placeholder or something like that. But if it needs to stay I'm OK to ease-up customization of applications, I'm OK with that.

> Then we would be able to perform this kind of override:
(...)
Yes, i know that it may be counterintuitive, but this would allow us to avoid the template overrides.

This would need to be documented in a subsection here: https://docs.decidim.org/en/v0.27/customize/images (You can do it in other PR if you want)

Reply:
> This would need to be documented in a subsection here: [docs.decidim.org/en/v0.27/customize/images](https://docs.decidim.org/en/v0.27/customize/images) (You can do it in other PR if you want)

I would do it in another PR, that would document ways of overriding Decidim ( classes, images, styles etc )